errors.go

```diff
@@ -100,3 +104,7 @@ func NewQueryError(err error, query string) *QueryError {
 func (e *QueryError) Error() string {
 	return fmt.Sprintf("%s: Query: %s", e.Err, e.Query)
 }
+
```

Comment:
This unwrap version is for support errors.As, some tests in warden 
![image](https://github.com/user-attachments/assets/1fae0858-d32f-404e-b51b-9f19a3b54b7a)

Since 1.21 casting to a nested errors won't result in the nested error, it could be fixed by double casting or use errors.As
![image](https://github.com/user-attachments/assets/3ed96418-c156-4635-9ad9-e3a5ed6014bb)